app/Controllers/Ws/ChatController.js

```diff
@@ -24,24 +23,46 @@ class ChatController {
       if (!user) {
         this.socket.broadcastToAll('error', 'User not found')
       }
-      let messages = await Message.create(data);
-      this.socket.broadcastToAll('chatMessage', messages);
+      await Message.create(data);
+      this.socket.broadcast('chatMessage', user)
   }
 
   async onGetChatMessage (data) {
-    console.log(data);
-    this.socket.broadcastToAll('message', data.receiver_id)
+    const user = await User.query().where('id', data.receiver_id);
+    this.socket.broadcast('message', user)
   }
 
- async onClose (error) {
-    // same as: socket.on('close')
-   this.socket.broadcastToAll('error',error)
+  async onSendChannelMessage (data) {
+    const { channel_id } = data;
+    const channel = await Channel.query().where('id', channel_id);
+    if (!channel) {
+      this.socket.broadcastToAll('error', 'No such channel exists');
+    }
+    const channel_users = channel.users().fetch();
+    await Message.create(data);
+    this.socket.broadcastToAll('message', channel_users);
   }
 
-  async onError (error) {
-    // same as: socket.on('error')
-    this.socket.broadcastToAll('error',error)
+  async onGetChannelMessage(data) {
+    const { channel_id } = data;
+    const channel = await Channel.query().where('id', channel_id);
+    if (!channel) {
+      this.socket.broadcastToAll('error', 'No such channel exists');
+    }
+    const channel_users = channel.users().fetch();
+    this.socket.broadcastToAll('message', channel_users);
   }
+
+//   async onClose (error) {
```

Comment:
@encodedBicoding Please enable the eror function on this controller

Reply:
this was what was causing that packet error.